core/controllers/story_editor.py

```diff
@@ -122,11 +121,8 @@ def put(self, topic_id, story_id):
             story_domain.StoryChange(change_dict)
             for change_dict in change_dicts
         ]
-        try:
```

Comment:
Don't remove this.

Reply:
The function [update_story](https://github.com/oppia/oppia/blob/develop/core/domain/story_services.py#L432) itself raises all the exceptions if anything goes wrong. If there isn't a commit message it will raise a value error and if the change_list is invalid, it will also raise an Exception. Hence the try block may not be needed here after all. I tried raising all possible exceptions but the exceptions were raised from `update_story()` not from this try block.

Reply:
Yeah, but those are general exceptions, which are used by this try - catch statement to raise InvalidInputException which are shown to the  user as an error message in the orange modal with the correct message, so it is needed.

Reply:
But the `InvalidInputException` from this block is not getting executed. I traced the exception statements and the exception is being raised from the original source, i.e `update_story()`. Do you know what to do here? Thanks!

Reply:
Did you try setting an invalid value to a field? Like a number to story title or something. I think that's when this  error is called. Same for topic editor.

Reply:
Now and in the future, whenever removing error-checking code please replace it with a comment which explains why there's no need to check. This will help reviewers as well, since we'll know that you've carefully thought about why the check is no longer necessary.

Reply:
@aks681 Yes i tried that but what I am trying to say is that the error is raised from the update_story function and the except block in this file is not getting executed.

Reply:
For example,to test the function for no commit message I tried editing the exception line to be like `raise InvalidInputException('errors')` but the error message from webapp is "no commit message provided" which is from the update_story function

Reply:
Commit message is a value error, so it won't be caught here. 
So, you tried setting invalid values to fields that raise validation error in the validate() functions in story_domain?
eg: Setting description to be a number and not a string.

Reply:
Ah yes you are right. Done!